node/metrics/useQuote.ts

```diff
@@ -14,7 +14,20 @@ type UseQuoteFieldsMetric = {
   quote_last_update: string
 }
 
-type UseQuoteMetric = Metric & { fields: UseQuoteFieldsMetric }
+export class UseQuoteMetric implements Metric {
+  public readonly description: string
+  public readonly kind: string
+  public readonly account: string
+  public readonly fields: UseQuoteFieldsMetric
+  public readonly name = B2B_METRIC_NAME
+
+  constructor(account: string, fields: UseQuoteFieldsMetric) {
+    this.account = account
+    this.fields = fields
+    this.kind = 'use-quote-graphql-event'
+    this.description = 'Use Quotation Action - Graphql'
+  }
+}
```

Comment:
From what I see here you're not using this `UseQuoteMetric` as a class in the code, never doing `new UseQuoteMetric`. It's not used as a type for object values.

In that case this shouldn't be a class. Classes are for when you really use instances, and are more rarely used in JS than in other languages. For your current usage this should just be a typescript type (or interface, which is the same thing) instead. So instead of having this class with an unused constructor this should be something like:

```suggestion
export interface UseQuoteMetric implements Metric {
  kind: 'use-quote-graphql-event'
  description: 'Use Quotation Action - Graphql'
  fields: UseQuoteFieldsMetric
}
```

Or even what this already was before :`type UseQuoteMetric = Metric & { fields: UseQuoteFieldsMetric }`. You'd just need to add the `kind` and `description` values to it as well.

Reply:
Same for the other metrics in this PR.

Reply:
My mistake! I forgot to use the new. The others options is not working because, using `type` or `interface` I need to declare the name, kind and description properties twice and repeat the same string. If you look at the code in the master branch, you can find the duplicated string for `use-quote-graphql-event` and others kind events.

Reply:
I'd prefer the duplication to using classes without need, as it's really not that common in JS. 

But I believe that if you use `interface + extends` you shouldn't need to repeat the fields that are already defined in the interface.

Reply:
I tested, and it hasn't worked. I don't know if I did something wrong. [Test](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAQgJjgfQLIFEAqAlAkgYWQDkBBDGAXhgHIAjBWgWggFcBLKAU0dpYE9OAJxCCA5owAmAQyhTqAKHlswXQQDMpwTjFScogtsBgBveTBiDOUieAA2fGJtAsVALhjQDYUWYtWbYPYwANbKEu6eyj7mltZ2DhKcEMAGAA5QbOAR+lG+sQFBYFIAtpzuUHypnCBq8EhoWHiEpBjyAL6KFVUwAKoQnACKLCBcAGJsnLYSELo5RlSm5gCOw1zIbOEeOd6+KyOcyJZqQpxgWshFpdle0TAiiYLIaiLF65uRO8urBylWGeDIaRca65L77ZAsfqAmRlLY3Xy-GQiZCcYpSNi2EGfGCQoQotEYrG3UDQZBaFR4jZE3y8ASPESiN7UsFrWxSUksVJA2EfHwdJQU9SabR9QbfWYGIycAAeXDA0x0eklJl8oXl7mouMYey4jFEgikqQAFktbIxOAA3U5QBTmRLJNL-MAa0UwIYjGSZMAwEjAJ0wRgwADiBuNpttMDUEymEHcovdY2j0wlhnailAkFgXU4EhTwDj-QTnDzlBV5nMThALigGsr1eoABpfOYo5Npu5FuXzLS8QymTQdZwAHQMqRgNgAL094EbzfLJKgZOtlM2mv2Q4X+GXglnXZZBypNG+Q42u73g8OnGOljOB0usLXXCHRxOt6IJU4Z679zxz0Er0PahBAeUYXlwCQv3LC9ESdaFgQHY8YK9AARGFIP3CEoW5dwwE4AB3GBUK4AAKABKIcoBAXAAGUAHlqO2UQyKbPcYBg5FUXRTEEPXdjBHQAlbHQnF+keTjCSPUSBK44SLzZDkuRhDVByHeSoB6RSuEgtoWJgfkMwgEBbGHWwQCY6hs1zJVDEbGBLLzUj03AaA2PZKASwWOc6zcGhvJtXSWyTWMyz3Ht6TEftqBU0dxynJ1hIXJdBUixCQGgLdBVk74UvXU8Aqg7KXxvc57w1Y8itOLR31KYSf0eP8ANXYChFA-9wKy8EkIBbCeKfLqwCIz98ow3E4NhXCCMGsiKKouiGJuZi53MPj8S45TENiSj+ME4TcTEwSyqknbhpgOS3IhTSHxUtSNO5bTdLaRwIF6QtxWs4AnMgIyTLM4jqGANy81sgHoAcoA)

Reply:
I am using classes because of this scenario. If you see, we have the interface and other events that need to have the same structure, so for me, the OOP fits with the necessity. I didn't find the restriction in the documentation don't use [classes](https://www.typescriptlang.org/docs/handbook/2/classes.html).

Reply:
Oh I get it now, it's for the fields that are defined witha single possible value, they still need to be defined when creating the object itself, not when defining the extended interface.

It's really fine to have the duplication in that case. There's only one place where this object is created, so this "repetition" is only between the type and that single use case. If there were more places creating these objects you could create a helper function to build it so that it added those fixed values for you. But having the repetition between type and value is preferrable to using a class/constructor just for this.

Reply:
@Rudge there's no restriction on not using classes, but it's really weird to use classes in JS, especially without functions. Creating objects directly is the expected and widely used pattern. You won't see many `new ` calls in JS repos.